app/src/main/java/com/example/cocktails/presentation/fragments/details/CocktailDetailsFragment.kt

```diff
@@ -0,0 +1,162 @@
+    private val cocktailDetailsViewModel: CocktailDetailsViewModel by hiltNavGraphViewModels(R.id.cocktailDetailsFragment)
+    private var argId: Int = 1
+    private var argFavorite = false
+    private var argEmail: String = ""
+    private var cocktailList: List<Cocktail> = mutableListOf()
+
+    private val stringBuilderIngredient = StringBuilder()
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View {
+        argId = args.argDetails
+        argFavorite = args.argsFavorite
+        argEmail = args.argEmail
+        _binding = FragmentCocktailDetailsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initObservers()
+    }
+
+    private fun initObservers() {
+        cocktailDetailsViewModel.cocktailsDetails.observe(viewLifecycleOwner) { resource ->
+            when (resource) {
+                is Resource.Success -> {
+                    binding.progressBar.visibility = View.GONE
+                    val cocktailDetails = resource.data
+                    cocktailDetailsMap = getNonNullFields(cocktailDetails)
+                    createList(cocktailDetails)
+                    setDetails()
+                    initUI()
+                }
+
+                is Resource.Loading -> {
+                    binding.progressBar.visibility = View.VISIBLE
+                }
+
+                is Resource.Error -> {
+                    binding.progressBar.visibility = View.GONE
+
+                    val builder: AlertDialog.Builder = AlertDialog.Builder(context)
+                    builder.setMessage(resources.getString(R.string.filter_error, resource.message))
+                        .setTitle("Error")
+                        .setPositiveButton("OK") { _, _ ->
+                        }
+
+                    val dialog: AlertDialog = builder.create()
+                    dialog.show()
+                }
+            }
+        }
+
+        cocktailDetailsViewModel.fetchAllCocktailDetails(argId)
+
+    }
+
+    private fun createList(cocktailDetails: List<CocktailDetails>) {
+        cocktailList = cocktailDetails.map { cocktailDetailsItem ->
+            Cocktail(
+                id = cocktailDetailsItem.idDrink,
+                name = cocktailDetailsItem.strDrink,
+                image = cocktailDetailsItem.strDrinkThumb,
+                alcoholic = cocktailDetailsItem.strAlcoholic,
+                favorite = argFavorite,
+                email = argEmail
+            )
+        }
+    }
+
+    private fun initUI() {
+        Glide.with(binding.cocktailImage.context)
+            .load(cocktailDetailsMap?.get("strDrinkThumb"))
+            .into(binding.cocktailImage)
+
+        binding.cocktailName.text = cocktailDetailsMap?.get("strDrink")
```

Comment:
Extract all these keys in constant variables in a companion object in this fragment